lib/improver/tests/wxcode/wxcode/test_wxcode_utilities.py

```diff
@@ -162,5 +310,151 @@ def test_list_of_lists(self):
             self.assertEqual(val, 'a')
 
 
+class Test_update_daynight(IrisTest):
+    """Test updating weather cube depending on whether it is day or night"""
+
+    def setUp(self):
+        """Set up for update_daynight class"""
+        self.wxcode = np.array(list(WX_DICT.keys()))
+        self.wxmeaning = " ".join(WX_DICT.values())
+        self.cube_data = np.array([[
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+            [3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3],
+            [5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5],
+            [6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6],
+            [7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7],
+            [8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8],
+            [10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10],
+            [14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14],
+            [15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15],
+            [17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17],
+            [18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18],
+            [20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20],
+            [23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23],
+            [26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 26],
+            [27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27],
+            [29, 29, 29, 29, 29, 29, 29, 29, 29, 29, 29, 29, 29, 29, 29, 29]]])
+
+    def test_basic(self):
+        """Test that the function returns a weather code cube."""
+        cube = set_up_wxcube()
+        result = update_daynight(cube)
+        self.assertIsInstance(result, Cube)
+        self.assertEqual(result.name(), 'weather_code')
+        self.assertEqual(result.units, Unit("1"))
+        self.assertArrayEqual(result.attributes['weather_code'], self.wxcode)
+        self.assertEqual(result.attributes['weather_code_meaning'],
+                         self.wxmeaning)
+
+    def test_raise_error_no_time_coordinate(self):
+        """Test that the function raises an error if no time coordinate."""
+        cube = set_up_wxcube()
+        cube.coord('time').rename('nottime')
+        msg = "cube must have time coordinate"
+        with self.assertRaisesRegex(CoordinateNotFoundError, msg):
+            update_daynight(cube)
+
+    def test_wxcode_updated(self):
+        """Test Correct wxcodes returned for cube."""
+        cube = set_up_wxcube()
+        cube.data = self.cube_data
+        # Only 1,3,10, 14, 17, 20, 23, 26 and 29 change from day to night
+        expected_result = np.array([[
```

Comment:
Just a thought, it might be worth factoring out this expected_result into the setUp so the big array isn't repeated so much.

Reply:
The expected result is subtly different in each of the tests.

Reply:
That's ok then, leave them where they are.